NuKeeper.Gitlab/GitlabSettingsReader.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Linq;
+using NuKeeper.Abstractions;
+using NuKeeper.Abstractions.CollaborationPlatform;
+using NuKeeper.Abstractions.Configuration;
+using NuKeeper.Abstractions.Formats;
+
+namespace NuKeeper.Gitlab
+{
+    public class GitlabSettingsReader : ISettingsReader
+    {
+        private const string GitLabTokenEnvironmentVariableName = "NuKeeper_gitlab_token";
+        private const string UrlPattern = "https://gitlab.com/{username}/{projectname}.git";
+
+        public Platform Platform => Platform.GitLab;
+
+        public bool CanRead(Uri repositoryUri)
+        {
+            return repositoryUri?.Host.Contains("gitlab", StringComparison.OrdinalIgnoreCase) == true;
+        }
+
+        public void UpdateCollaborationPlatformSettings(CollaborationPlatformSettings settings)
+        {
+            UpdateTokenSettings(settings);
+            settings.ForkMode = settings.ForkMode ?? ForkMode.PreferFork;
+        }
+
+        private static void UpdateTokenSettings(CollaborationPlatformSettings settings)
+        {
+            var envToken = Environment.GetEnvironmentVariable(GitLabTokenEnvironmentVariableName);
+            settings.Token = Concat.FirstValue(envToken, settings.Token);
+        }
+
+        public RepositorySettings RepositorySettings(Uri repositoryUri, string targetBranch = null)
+        {
+            if (repositoryUri == null)
+            {
+                throw new NuKeeperException(
+                    $"The provided uri was is not in the correct format. Provided null and format should be {UrlPattern}");
+            }
+
+            // Assumption - url should look https://gitlab.com/{username}/{projectname}.git";
```

Comment:
Here we can't really be sure, because GitLab can also be hosted on premises. But we have some structure checks following later. Taking similar approach as in Bitbucket or Azure implementation.